cmd/login/runner.go

```diff
@@ -107,26 +108,31 @@ func (r *runner) run(ctx context.Context, cmd *cobra.Command, args []string) err
 	return nil
 }
 
-func (r *runner) tryToGetCurrentContext(ctx context.Context) (string, error) {
+func (r *runner) tryToGetCurrentContexts(ctx context.Context) (string, string, error) {
 	config, err := r.commonConfig.GetConfigAccess().GetStartingConfig()
 	if err != nil {
-		return "", microerror.Mask(err)
+		return "", "", microerror.Mask(err)
 	}
-	return config.CurrentContext, nil
+	contextOverride := r.commonConfig.GetContextOverride()
+	if _, ok := config.Contexts[contextOverride]; !ok {
+		contextOverride = ""
+	}
+	return config.CurrentContext, contextOverride, nil
 }
 
 func (r *runner) setLoginOptions(ctx context.Context, args *[]string) {
-	originContext, err := r.tryToGetCurrentContext(ctx)
+	originContext, contextOverride, err := r.tryToGetCurrentContexts(ctx)
 	if err != nil {
 		fmt.Fprintln(r.stdout, color.YellowString("Failed trying to determine current context. %s", err))
 	}
 	r.loginOptions = LoginOptions{
 		originContext:           originContext,
+		contextOverride:         contextOverride,
 		isWCClientCert:          len(r.flag.WCName) > 0,
 		selfContained:           len(r.flag.SelfContained) > 0 && !(len(r.flag.WCName) > 0),
 		selfContainedClientCert: len(r.flag.SelfContained) > 0 && len(r.flag.WCName) > 0,
 	}
-	r.loginOptions.switchToContext = r.loginOptions.isWCClientCert || !(r.loginOptions.selfContained || r.flag.KeepContext)
+	r.loginOptions.switchToContext = r.loginOptions.contextOverride == "" && (r.loginOptions.isWCClientCert || !(r.loginOptions.selfContained || r.flag.KeepContext))
```

Comment:
This formula was already confusing before, maybe now is the time where we should separate it and add some comments wdyt?

Reply:
Restructured the conditions and used more descriptive names to make them more self-explanatory. Is is better or should I try a different approach?